content/en/blog/_posts/2021-04-08-cronjob-reaches-ga/index.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: blog
+title: 'CronJob Reaches GA'
+date: 2021-04-09
+slug: cronjob-reaches-ga
+---
+
+ **Authors:** Alay Patel (Red Hat), and Maciej Szulik (Red Hat)
+
+Since last summer, we have been working on a new CronJob Controller (Controller
+V2) implementation which initially landed in 1.20 as an alpha feature. In 
+kubernetes 1.21 the new controller is turned on by default, and the CronJob 
+resource is promoted to general availability (GA) (group version: `batch/v1`). This blog
+post walks through the driving forces behind this new development, gives a 
+brief description of a controller design in core kubernetes, and explains what
+users will gain with this new controller.
+
+The driving force behind promoting the API was this [proposal](https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/1635-prevent-permabeta#sig-apps). Its acceptance
+prevents APIs from being stuck in a permanent beta state. Over the years the
+old CronJob controller had received healthy feedback from the community and
+it had some well-known [issues](https://github.com/kubernetes/kubernetes/issues/82659).
+If this API was to be supported as GA, the controller would need a severe re-work.
+Instead, the SIG Apps community decided to introduce a new controller to
+gradually replace the old one. 
+
+## How do controllers work?
+
+The old controller does a periodic poll and sweep of all the objects in order 
+to act on them. It is a single worker implementation that gets all CronJobs 
+every 10 seconds, iterates over each one of them, and syncs them to their 
+desired state. This was the default way of doing things almost 5 years ago when
+the controller was initially written. In hindsight, we can certainly say that 
+such an approach can overload the API server at scale.
+
+These days, every core controller in kubernetes must follow the guidelines 
+described in this [document](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/controllers.md). 
+Among many details, it prescribes using shared informers to “receive
```

Comment:
There's a blog article on shared informers than I recommend linking to, unless it's no longer accurate:
```suggestion
Among many details, that document prescribes using
[shared informers](https://www.cncf.io/blog/2019/10/15/extend-kubernetes-via-a-shared-informer/)
to “receive
```